### DISCUSSION TITLE
Terrain details and trees disappear when adding "NatureRenderer" component to terrain with TVE prefabs

### DISCUSSION THREAD

**Discussion by mroshaw** [visualdesigncafe/nature-renderer#21] [Help] [answered]
Hi! I'm using Unity 2022.3.17f1, HDRP 14.0.9, Nature Renderer 2022.0.3. All foliage prefabs have been converted with The Vegetation Engine 12.5.0 and placed using MicroVerse 1.7.23.

I've installed Nature Renderer 2022, and applied all updates across all components. I'm following the Quick Start guide to add NR to an existing terrain.

When I add the Nature Renderer component, all terrain details and trees immediately disappear. They are not visible in the editor, nor when entering run mode. The component is picking up the prefabs for the existing detail and tree prefabs, and there are no errors or warnings shown. There is no warning or button to "Patch shaders", presumably because the TVE shaders are fully supported.

I'm not sure where to look next - any ideas why my details and trees are not being shown at all?

**Comment by mroshaw** [comment, marked as answer]
Ah, to get it all working with The Vegetation Engine, I had to:

Go to Window > BOXOPHOBIC > The Vegetation Engine > Shader Manager
Under "Render Engine Support", select "Nature Renderer (Procedural Instancing)

And that's it - terrain details and trees appear correctly!

**Reply by natfoth** [reply]
Thank you. I wish the was a little more obvious, responding to give it some more search traffic.
